flytekit/control_plane/component_nodes.py

```diff
@@ -0,0 +1,137 @@
+import logging as _logging
+from typing import Dict
+
+from flytekit.common.exceptions import system as _system_exceptions
+from flytekit.control_plane import identifier as _identifier
+from flytekit.models import task as _task_model
+from flytekit.models.core import workflow as _workflow_model
+
+
+class FlyteTaskNode(_workflow_model.TaskNode):
+    def __init__(self, flyte_task) -> "flytekit.control_plane.tasks.task.FlyteTask":
+        self._flyte_task = flyte_task
+        super(FlyteTaskNode, self).__init__(None)
+
+    @property
+    def reference_id(self) -> _identifier.Identifier:
+        """A globally unique identifier for the task."""
+        return self._flyte_task.id
+
+    @property
+    def flyte_task(self) -> "flytekit.control_plane.tasks.task.FlyteTask":
+        return self._flyte_task
+
+    @classmethod
+    def promote_from_model(
+        cls,
+        base_model: _workflow_model.TaskNode,
+        tasks: Dict[_identifier.Identifier, _task_model.TaskTemplate],
+    ) -> "FlyteTaskNode":
+        """
+        Takes the idl wrapper for a TaskNode and returns the hydrated Flytekit object for it by fetching it with the
+        FlyteTask control plane.
+
+        :param base_model:
+        :param tasks:
+        """
+        from flytekit.control_plane.tasks import task as _task
+
+        if base_model.reference_id in tasks:
+            task = tasks[base_model.reference_id]
+            _logging.info(f"Found existing task template for {task.id}, will not retrieve from Admin")
+            flyte_task = _task.FlyteTask.promote_from_model(task)
+            flyte_task._has_registered = True
```

Comment:
we probably don't need any `_has_registered` flags.  Because the serialize flow now [checks](https://github.com/flyteorg/flytekit/blob/master/flytekit/clis/sdk_in_container/serialize.py#L173) for specific classes, we don't have to worry about the stuff we read from Admin getting reregistered back to admin.